src/span.h

```diff
@@ -78,8 +96,19 @@ class Span
      * To prevent surprises, only Spans for constant value types are supported when passing in temporaries.
      * Note that this restriction does not exist when converting arrays or other Spans (see above).
      */
-    template <typename V, typename std::enable_if<(std::is_const<C>::value || std::is_lvalue_reference<V>::value) && std::is_convertible<typename std::remove_pointer<decltype(std::declval<V&>().data())>::type (*)[], C (*)[]>::value && std::is_convertible<decltype(std::declval<V&>().size()), std::size_t>::value, int>::type = 0>
-    constexpr Span(V&& v) noexcept : m_data(v.data()), m_size(v.size()) {}
+    template <typename V>
+    constexpr Span(V& other SPAN_ATTR_LIFETIMEBOUND,
+        typename std::enable_if<!is_Span<V>::value &&
+                                std::is_convertible<typename std::remove_pointer<decltype(std::declval<V&>().data())>::type (*)[], C (*)[]>::value &&
+                                std::is_convertible<decltype(std::declval<V&>().size()), std::size_t>::value, std::nullptr_t>::type = nullptr)
+        : m_data(other.data()), m_size(other.size()){}
+
+    template <typename V>
+    constexpr Span(const V& other SPAN_ATTR_LIFETIMEBOUND,
+        typename std::enable_if<!is_Span<V>::value &&
+                                std::is_convertible<typename std::remove_pointer<decltype(std::declval<const V&>().data())>::type (*)[], C (*)[]>::value &&
+                                std::is_convertible<decltype(std::declval<V&>().size()), std::size_t>::value, std::nullptr_t>::type = nullptr)
```

Comment:
Nit: `declval<const V&>`. Not that I expect size() to differ between const and non-const objects...

Reply:
Fixed, thanks.